sourcegraph/sourcegraph
# Code Insights: Connect UI and compute insight, assemble creation UI page

Based on https://github.com/sourcegraph/sourcegraph/pull/38739
Blocked (live preview component) by https://github.com/sourcegraph/sourcegraph/pull/38125
Closes https://github.com/sourcegraph/sourcegraph/issues/37965

## Background

<img width="400" alt="Screenshot 2022-07-14 at 17 49 40" src="https://user-images.githubusercontent.com/18492575/179011215-9ad00078-fa8b-4117-95ed-1a965e8475f3.png">

This PR integrates live preview and creation form components. A few things to notice 
- Compute-powered insight creation UI has old designs (like all other creation forms at the moment). FYI @AlicjaSuska all redesign work won't be included in the first version of compute insight track and will be addressed later in this [issue](https://github.com/sourcegraph/sourcegraph/issues/37218)
- Live preview doesn't have a timeframe field. It turned out this isn't so trivial to implement. Because our after/before filters may have different (sometimes complex to parse) values for example, `after:"6 months ago"`.  Taking into account our deadline for the first version I suggest moving this out for the second iteration over compute insight.
- Series don't have colour picker and we don't colour bar in series colour, see this issue https://github.com/sourcegraph/sourcegraph/issues/38832 for more details

## Todo
- [x] Add actions UI to the compute creation UI page
- [x] Update GQL API class `createInsight` method with compute powered insight
- [x] Rebase the main and integrate live preview component 
- [x] Testing creation UI locally and with k8s

## Test plan
- Go to the `/insights/create/group-results` URL. 
- Make sure that you can create a compute-powered insight through the newly created compute creation UI page and see a correct live preview result.



## App preview:

- [Web](https://sg-web-vk-connect-ui-and-compute-insight-4cig.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-hxgcppiycc.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
